### ISSUE TITLE
[Docs]: Mysterious exports in app/root.jsx

### ISSUE THREAD

**Issue by tschaub** [remix-run/react-router#12183]
### Describe what's incorrect/missing in the documentation

In the manual installation instructions, it suggests creating an `app/root.jsx` file that exports `Layout`, `HydrateFallback`, `ErrorBoundary`, and a `default` component.

https://github.com/remix-run/react-router/blob/31a9ad8477544f2d41cc41e49422c08c286ab88e/docs/start/installation.md?plain=1#L63-L91

It looks to me like the `Layout` export is used when rendering the root route, but that the `default` export (named `Root` in the module) is unused (the "Hello, world!" content is not rendered).

Later, [the docs](https://github.com/remix-run/react-router/blob/31a9ad8477544f2d41cc41e49422c08c286ab88e/docs/start/routing.md#root-route) say "Every route in `routes.ts` is nested inside the special `app/root.tsx` module."

This leaves me with a few questions:
 1. Should the `Layout` component in `app/root.jsx` render the `children` prop (instead of rendering `<Outlet/>`)?
 2. Should the `default` export from `app/root.jsx` render `<Outlet/>`?
 3. Could the docs describe more specifically how the `default` and other named exports from `app/root.jsx` (especially `Layout`) are used (instead of just saying routes are "nested" in the special root module)?

I really appreciate the manual instructions and understand that this `dev` branch is still a work in progress.

**Comment by oliverjam** [comment]
Yeah I found this confusing. It looked like the default `Root` export wasn't used anywhere, so I deleted it. This seemed fine in dev, but when I pre-rendered to HTML my pages were missing the stuff from `Layout` (i.e. just bare page HTML like `<h1>About page</h1>`.

I assume the presence of the default export is required for `Layout` to work somehow?

**Comment by brookslybrand** [comment]
Yep, this doc didn't explain it well at the time

The `Layout` export is unique to `root.tsx` and is documented [here](https://reactrouter.com/explanation/special-files#layout-export)
